### PR TITLE
No advertising for Internet Explorer

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

This PR adds a new check `isUnsupportedBrowser` to the set of commercial features. We use this new flag to disable advertising if set to `true`, which occurs when an Internet Explorer user agent is detected.

Additionally, this adds some tests to check this logic against some sample user agents.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

The commercial code runs inconsistently on Internet Explorer, with some features supported and others badly broken. Since IE is not a great contributor to advertising revenue (lots of the ads don't load in the first place), it makes sense to consider it unsupported from a commercial perspective and not attempt to show ads at all.

### Tested

- [X] Locally
- [ ] On CODE (optional) - **TODO**

### CODE REVIEW

File: static/src/javascripts/lib/detect.js [hunk 1]
```diff
@@ -270,7 +270,7 @@ const adblockInUse = getAdblockInUse();
 
 const getReferrer = () => document.referrer || '';
 
-const getUserAgent = (() => {
+const getUserAgent = () => {
```
Comment (chrislomaxjones) [inline static/src/javascripts/lib/detect.js:273]: We change the type of this here from `string` to `() => string`. This makes it much easier to test, and I think more flexible about where you use it.

I assume the original reason for calling the function immediately was to only compute the user agent once and not every time you wanted it, so I've included a `userAgent` that is computed immediately and exported. This is used in two JS files than I'm keen not to mess with too much.

File: static/src/javascripts/projects/common/modules/commercial/commercial-features.ts [hunk 2]
```diff
@@ -30,6 +30,23 @@ function adsDisabledLogger(
 	}
 }
 
+/**
+ * Determine whether current browser is a version of Internet Explorer
+ */
+const isInternetExplorer = () => {
```
Comment (arelra) [inline static/src/javascripts/projects/common/modules/commercial/commercial-features.ts:36]: Might this be better in `detect.js`?

File: static/src/javascripts/projects/common/modules/commercial/commercial-features.spec.ts [hunk 3]
```diff
@@ -117,6 +137,62 @@ describe('Commercial features', () => {
 			expect(features.adFree).toBe(true);
 			expect(features.dfpAdvertising).toBe(false);
 		});
+
+		describe('In browser', () => {
```
Comment (arelra) [inline static/src/javascripts/projects/common/modules/commercial/commercial-features.spec.ts:141]: Yay for adding tests!

**Review by arelra — APPROVED** [review]
🎉  Looks good! Just a minor comment about where `isInternetExplorer()` should live.

File: static/src/javascripts/projects/common/modules/commercial/commercial-features.ts (hunk 2 above)
Comment (chrislomaxjones) [reply static/src/javascripts/projects/common/modules/commercial/commercial-features.ts:36]: Great shout

Comment (chrislomaxjones) [reply static/src/javascripts/projects/common/modules/commercial/commercial-features.ts]: Done 🙂

**Comment by mxdvl** [comment]
This is probably the nicest thing we've done for IE users in a long time!

**Pull request merged**